mlrun/datastore/sources.py

```diff
@@ -429,6 +429,59 @@ def to_spark_df(self, session, named_view=False):
         return df
 
 
+class SnowflakeSource(BaseSourceDriver):
+    support_spark = True
+    support_storey = False
+
+    def __init__(
+        self,
+        name: str = "",
+        attributes: Dict[str, str] = None,
+        key_field: str = None,
+        time_field: str = None,
+        schedule: str = None,
+        query: str = None,
+        url: str = None,
+        user: str = None,
+        password: str = None,
+        database: str = None,
+        schema: str = None,
+        warehouse: str = None,
+    ):
+        self.query = query
+        self.attributes = attributes
+        self.key_field = key_field
+        self.time_field = time_field
+        self.schedule = schedule
+        self.url = url
+        self.user = user
+        self.password = password
+        self.database = database
+        self.schema = schema
+        self.warehouse = warehouse
+
+        super().__init__(
+            name,
+            attributes=attributes,
+            key_field=key_field,
+            time_field=time_field,
+            schedule=schedule,
+        )
+
+    def get_spark_options(self):
+        return {
+            "format": "net.snowflake.spark.snowflake",
+            "query": self.query,
+            "sfURL": self.url,
+            "sfUser": self.user,
+            "sfPassword": self.password,
+            "sfDatabase": self.database,
+            "sfSchema": self.schema,
+            "sfWarehouse": self.warehouse,
+            "application": f"Iguazio-{os.getenv('SNOWFLAKE_APPLICATION', 'application')}",
```

Comment:
why are you add Iguazio here? Why not include igauzio in the env variable?
Also shouldn't the default value be snowflake_application or something more detailed than application?